File: ArrayOperation.c
/* 
    Author : Md. Abid Hasan Roni
    Dept of CSE
    Faridpur Engineering College
*/

//Array Travers , Insert , Delete and Searching 
#include <stdio.h>

//Traversing Function
int Travers(int N[],int s) 
{
    int i;
    printf("Traversing is : ");
    for(i=0;i<s; i++)
        printf("%d   ",N[i]);
    printf("\n");
    return Lab1(N,s);
}

//Insertion Function
int Insert(int L[],int s) 
{
    int p,j=s,v; // p=position , v = value
    printf("Enter The Insert position : ");
    scanf("%d",&p);
    printf("Enter the insert Value :");
    scanf("%d",&v);
    while (j>=p-1)
    {
        L[j+1]=L[j];
        j=j-1;
    }
    L[p-1]=v;
    s=s+1;
    return Lab1(L,s);
}

//Deleting Function
int Delete(int N[],int s) 
{
    int i,j,v;//v = deleting value
    printf("Enter Delete value : ");
    scanf("%d",&v);
    for(i=0;i<s;i++)
    {
        if(N[i]==v)
        {
            for(j=i;j<s;j++)
                N[j]=N[j+1];
        }
    }
    s=s-1;
    return Lab1(N,s);
}

//Searching Function
int Search(int N[],int s) 
{
    int v ,i, found=0; //v= searching value p=found position
    printf("Enter Searching value : ");
    scanf("%d",&v);
    for(i=0;i<s;i++)
    {
        if(N[i]==v)
        {
            printf("Found in position : %d\n",i+1);
            found++;
        }
    }   
    if(found>0)
        printf("Total Found : %d\n",found);
    else 
        printf("Search value not found \n");
    return Lab1(N,s);
}

//Working Function
int Lab1(int N[],int s) 
{
    int option;
    printf("\t\tPlease Choise an option : \n\t1.Travers Array \n\t2.Intert an Element \n");
    printf("\t3.Delete an element \n\t4.Seach an element \n\t\t Press 0 to Exit\n");
    scanf("%d",&option);
    if(option==1)
        Travers(N,s);
    else if(option==2)
        Insert(N,s);
    else if(option==3)
        Delete(N,s);
    else if(option==4)
        Search(N,s);
    else if(option==0)
        return 0;
    else 
        printf("Please Choise a Valid option \n");
}

// This is The main Function
int main() 
{
    int size;
    printf("Enter The Array Size : ");
    scanf("%d", &size);\
    int A[size],i;
    printf("Enter The Element of the Array :");
    for(i=0;i<size;i++)
    {
        scanf("%d",&A[i]);
    }
    Lab1(A,size);
    return 0;
}

File: ArrayOperation.java
//Array Operation
import java.util.*;
public class Main{
    public static void main(String[] args) {
        Scanner in=new Scanner(System.in);
        System.out.println("Enter Array Size : ");
        int i,n=in.nextInt();
        int[] A=new int[100]; 
        System.out.println("Enter Your Array Element : ");
        for(i=0;i<n;i++)
            A[i]=in.nextInt();
        System.out.println("The Array Elements is : ");
        for(i=0;i<n;i++)
            System.out.print(A[i]+" ");
        Array obj=new Array();
        obj.A(A,n);
    }
}
class Array{
    Scanner in=new Scanner(System.in);
    int A(int[] A,int n){
        System.out.println("\nPlease select an option : \n1.Insert\n2.Delete\n3.Search \nPress 0 to Exit");
        int i, option=in.nextInt();
        if(option==1){
            n=n+1;
            System.out.println("Enter Insert Possition & value :");
            int p=in.nextInt();
            int value=in.nextInt();
            for(i=n-1;i>=p-1;i--)
                A[i]=A[i-1];
            A[p-1]=value;
            System.out.flush();
            System.out.println("The Array Elements is : ");
            for(i=0;i<n;i++)
                System.out.print(" "+A[i]+" ");
            return A(A,n);
        }
        else if(option==2){
            n=n-1;
            System.out.println("Enter Delete Position : ");
            int p=in.nextInt();
            for(i=p-1;i<n;i++)
                A[i]=A[i+1];
            System.out.println("The Array Elements is : ");
            for(i=0;i<n;i++)
                System.out.print(" "+A[i]+" ");
            return A(A,n);
        }
        else if(option==3){
            System.out.println("Enter The search Element : ");
            int value=in.nextInt();
            int found=0;
            for(i=0;i<n;i++){
                if(A[i]==value){
                    found =1;
                    break;
                }
            }
            if(found==1)
                System.out.println("Data found in Location : "+i);
            else 
                System.out.println("Data not Found ");
            return A(A,n);
        }
        return 0;
    }
}